src/workflow/invocation.rs

```diff
@@ -90,26 +96,15 @@ impl TryFrom<Ipld> for Batch {
     type Error = anyhow::Error;
 
     fn try_from(ipld: Ipld) -> Result<Self, Self::Error> {
-        match ipld {
-            Ipld::Map(assoc) => {
-                let mut batch = BTreeMap::new();
-
-                let flow = assoc
-                    .iter()
-                    .try_for_each(|(key, value)| match Task::try_from(value) {
-                        Ok(task) => {
-                            batch.insert(TaskLabel(key.to_string()), task);
-                            ControlFlow::Continue(())
-                        }
-                        _ => ControlFlow::Break("invalid IPLD Task"),
-                    });
-
-                match flow {
-                    ControlFlow::Continue(_) => Ok(Batch(batch)),
-                    ControlFlow::Break(reason) => bail!(reason),
-                }
-            }
-            _ => bail!("Can only convert from a map"),
-        }
+        let map = from_ipld::<BTreeMap<String, Ipld>>(ipld)?;
+        let flow = map
+            .iter()
+            .try_fold(BTreeMap::new(), |mut acc, (key, value)| {
+                let task = Task::try_from(value)?;
+                acc.insert(TaskLabel::new(key.to_string()), task);
+                Ok::<_, anyhow::Error>(acc)
+            })?;
```

Comment:
Oh yeah, that's definitely cleaner ✨ I'm still getting used to what the right patterns for this stuff is. Glad to see that folds are idiomatic!

Reply:
yep. the folds/try_folds over iters are good, just need that silly type annotation!

Reply:
Is the `ControlFlow` stuff not worth it I guess? Probably fast enough just to return a bunch of `Err` in quick succession if you end up on that track?

Reply:
I didn't see the need here as we can bubble up. I'd use it if there's a result I really want to break with and use, cause then you have to match on the ControlFlow ops at the call site.